lib/thread.go

```diff
@@ -0,0 +1,21 @@
+package lib
+
+import (
+	"context"
+
+	"golang.org/x/sync/errgroup"
+)
+
+// ErrorGroup resolves all go routines. It fails at first error encountered
+func ErrorGroup(ctx context.Context, args ...func() error) error {
+	g, _ := errgroup.WithContext(ctx)
+
+	for _, fn := range args {
+		g.Go(fn)
```

Comment:
go routine?

Reply:
Seems more like a `Parallel.Invoke`, this method could even be called with itself as an argument...
Unless there is a way good way of separating any of the arg funcs as a goroutine or simple call, one way would be passing funcs as an array of an interface that explicitly identifies it